src/MessagePack.CodeGenerator/CodeAnalysis/TypeCollector.cs

```diff
@@ -626,13 +626,18 @@ void CollectObject(INamedTypeSymbol type)
             }
 
             // GetConstructor
+            IEnumerator<IMethodSymbol> ctorEnumerator = null;
             var ctor = type.Constructors.Where(x => x.DeclaredAccessibility == Accessibility.Public).SingleOrDefault(x => x.GetAttributes().Any(y => y.AttributeClass == typeReferences.SerializationConstructorAttribute));
```

Comment:
Of course the code generator (pure C # external class) can not call private constructor.